frame/asset-conversion/src/lib.rs

```diff
@@ -1141,6 +1141,45 @@ where
 
 		Ok(amount_in)
 	}
+
+	// If successful returns the amount out.
+	fn swap_exact_native_for_tokens(
+		sender: T::AccountId,
+		asset_id: T::AssetId,
+		amount_in: T::Balance,
+		amount_out_min: Option<T::AssetBalance>,
+		send_to: T::AccountId,
+		keep_alive: bool,
+	) -> Result<T::AssetBalance, DispatchError> {
+		ensure!(amount_in > Zero::zero(), Error::<T>::ZeroAmount);
+		if let Some(amount_out_min) = amount_out_min {
+			ensure!(amount_out_min > Zero::zero(), Error::<T>::ZeroAmount);
+		}
+		let mut path = sp_std::vec::Vec::new();
+		path.push(T::MultiAssetIdConverter::get_native());
+		path.push(T::MultiAssetIdConverter::into_multiasset_id(&asset_id));
+		let path = path.try_into().expect("MaxSwapPathLength must be greater than 1");
+
+		let amount_in = Self::native_to_asset(amount_in)?;
+
+		let amounts = Self::get_amounts_out(&amount_in, &path)?;
```

Comment:
I think some comments here (better, proper documentation on `fn native_to_asset` and `fn get_amounts_out`) would be helpful. I got a bit lost trying to read this (namely, I thought `native_to_asset` was perhaps doing a conversion, and thought there was a mistake until I realized it was just changing the type).

Reply:
Agree on `native_to_asset`, it should be named better. Any suggestions?